lib/android/app/src/main/java/com/reactnativenavigation/react/NavigationModule.java

```diff
@@ -159,12 +163,28 @@ private Navigator navigator() {
 
 	@NonNull
 	private LayoutFactory newLayoutFactory() {
-		return new LayoutFactory(activity(),
-                navigator().getChildRegistry(),
+
+		NavigationActivity layoutActivity = activity();
```

Comment:
This logic shouldn't be here, seems like it's more appropriate to `RootPresenter`.

Reply:
Right, but I'm not sure how can I access to reactContext in `RootPresenter`?

Reply:
Add reactContext as an argument to Navigator.setRoot, then pass it on when calling rootPresenter.setRoot